src/routes/token.rs

```diff
@@ -11,11 +11,18 @@ pub struct TokenResponse {
     pub id_token: String,
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct TokenMetadata {
+    pub client_id: String,
+    pub redirect_uri: String,
+}
+
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct TokenRequestBody {
     pub grant_type: String,
     pub code: String,
     pub redirect_uri: String,
+    pub client_secret: Option<String>,
```

Comment:
Already mentioned in DM, if the `client_secret` is present, also the `client_id` must be present. This can be an enum that, when deserialised, is flattened so that both properties appear in the top level.

Reply:
I guess alternatively we can make both optional and require either both to be present or none to be present.

Reply:
[912e6b5](https://github.com/KILTprotocol/opendid/pull/53/commits/912e6b5c5e523e86c91b4a477e70c954ae0683f8)